### PR TITLE
Add initial design proposal for nextgen applier, Dash

### PR DESCRIPTION
**Pull request by etsauer**
#### What does this PR do?
We had our first design session around the next major iteration of applier. Here's a proposal doc resulting from that meeting. It will be a living document for some time, so we're not expecting this PR to be perfect.

#### Who would you like to review this?
cc: @redhat-cop/openshift-applier

### CODE REVIEW

File: NEXTGEN_DESIGN_PROPOSAL.md [hunk 1]
```diff
@@ -0,0 +1,62 @@
...
+
+The working name for the project is "Dash" (because when you remove the words `openshift` and `applier`, you're left with `-`).
+
+## Design Philosophy
+
+TODO
+
+## Desired Feature List
+
+- Dash should take the form of a CLI tool
```
Comment (oybed) [inline NEXTGEN_DESIGN_PROPOSAL.md]: Do we want to mention anything about it being a library that can be consumed directly (in addition to the CLI option)?

**Comment by tylerauerbeck** [comment]
@etsauer So I like this as the start of the design proposal. I do have a few additional thoughts on what the inventory looks like though but not sure we want this to get stuck debating the inventory layout. Do we want to have these discussions elsewhere (a different PR, an issue, etc.) or are we okay having these held here prior to merge?

**Comment by oybed** [comment]
@tylerauerbeck IMHO I'd like to get this PR merged rather quickly and then accepting additional follow-up PRs. In that way, we can have a bit more focused conversations in the specific areas - instead of potentially holding up the entire write-up at first.

**Comment by etsauer** [comment]
@tylerauerbeck I would say if you have specific thoughts on what I've written so far, feel free to comment up the doc. But let's not consider merging this PR as "proposal complete".

File: NEXTGEN_DESIGN_PROPOSAL.md (hunk 1 above)
Comment (etsauer) [reply NEXTGEN_DESIGN_PROPOSAL.md]: @oybed done.

**Pull request merged**